Magento2/Sniffs/Legacy/LayoutSniff.php

```diff
@@ -0,0 +1,438 @@
+        'setHideLink',
+        'setHideRequiredNotice',
+        'setHtmlClass',
+        'setId',
+        'setImageType',
+        'setImgAlt',
+        'setImgHeight',
+        'setImgSrc',
+        'setImgWidth',
+        'setInList',
+        'setInfoTemplate',
+        'setIsCollapsed',
+        'setIsDisabled',
+        'setIsEnabled',
+        'setIsGuestNote',
+        'setIsHandle',
+        'setIsLinkMode',
+        'setIsPlaneMode',
+        'setIsTitleHidden',
+        'setIsViewCurrent',
+        'setItemLimit',
+        'setLabel',
+        'setLabelProperties',
+        'setLayoutCode',
+        'setLinkUrl',
+        'setListCollection',
+        'setListModes',
+        'setListOrders',
+        'setMAPTemplate',
+        'setMethodFormTemplate',
+        'setMyClass',
+        'setPageLayout',
+        'setPageTitle',
+        'setParentType',
+        'setControllerPath',
+        'setPosition',
+        'setPositioned',
+        'setRewardMessage',
+        'setRewardQtyLimitationMessage',
+        'setShouldPrepareInfoTabs',
+        'setShowPart',
+        'setSignupLabel',
+        'setSourceField',
+        'setStoreVarName',
+        'setStrong',
+        'setTemplate',
+        'setText',
+        'setThemeName',
+        'setTierPriceTemplate',
+        'setTitle',
+        'setTitleClass',
+        'setTitleId',
+        'setToolbarBlockName',
+        'setType',
+        'setUseConfirm',
+        'setValueProperties',
+        'setViewAction',
+        'setViewColumn',
+        'setViewLabel',
+        'setViewMode',
+        'setWrapperClass',
+        'unsetChild',
+        'unsetChildren',
+        'updateButton',
+        'setIsProductListingContext',
+    ];
+
+    /**
+     * @inheritdoc
+     */
+    public function register(): array
+    {
+        return [
+            T_INLINE_HTML
+        ];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        if ($stackPtr > 0) {
+            return;
+        }
+
+        $layout = simplexml_load_string($this->getFormattedXML($phpcsFile));
+
+        if ($layout === false) {
+            $phpcsFile->addError(
+                sprintf(
+                    "Couldn't parse contents of '%s', check that they are in valid XML format",
+                    $phpcsFile->getFilename(),
+                ),
+                $stackPtr,
+                self::ERROR_CODE_XML
+            );
+            return;
+        }
+
+        $this->testObsoleteReferences($layout, $phpcsFile);
+        $this->testObsoleteAttributes($layout, $phpcsFile);
+        $this->testHeadBlocks($layout, $phpcsFile);
+        $this->testOutputAttribute($layout, $phpcsFile);
+        $this->testHelperAttribute($layout, $phpcsFile);
+        $this->testListText($layout, $phpcsFile);
+        $this->testActionNodeMethods($layout, $phpcsFile);
+    }
+
+    /**
+     * Check for obsolete block references
+     *
+     * @todo missing test
```

Comment:
I haven't been able to find any `<reference>` in layouts, but `<referenceBlock>`. Could it be that this test is outdated?